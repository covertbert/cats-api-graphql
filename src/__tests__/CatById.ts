import { gql } from 'apollo-server'
import { createTestClient } from 'apollo-server-testing'

import server from '../apollo'

describe('CatsById integration tests', () => {
  it('queries a cat by its ID', async () => {
    const { query } = createTestClient(server)

    const CATS = gql`
      query {
        getCatById(id: 5) {
          name
          adaptability
          affectionLevel
          alternativeNames
          childFriendliness
          countryCode
          description
          dogFriendliness
          energyLevel
          experimental
          grooming
          hairless
          healthIssues
          hypoallergenic
          indoor
          intelligence
          lifeSpan
          natural
          origin
          rare
          rex
          shortLegs
          sheddingLevel
          socialNeeds
          strangerFriendly
          suppressedTail
          temperament
          vocalisation
          weight
        }
      }
    `

    const res = await query({ query: CATS })
    expect(res).toMatchSnapshot()
  })
})
