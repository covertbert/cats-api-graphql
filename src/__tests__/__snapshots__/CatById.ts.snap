// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatsById integration tests queries a cat by its ID 1`] = `
Object {
  "data": Object {
    "getCatById": Object {
      "adaptability": 5,
      "affectionLevel": 5,
      "alternativeNames": Array [
        "Domestic Shorthair",
      ],
      "childFriendliness": 4,
      "countryCode": "US",
      "description": "The American Shorthair is known for its longevity, robust health, good looks, sweet personality, and amiability with children, dogs, and other pets.",
      "dogFriendliness": 5,
      "energyLevel": 3,
      "experimental": false,
      "grooming": 1,
      "hairless": false,
      "healthIssues": 3,
      "hypoallergenic": false,
      "indoor": false,
      "intelligence": 3,
      "lifeSpan": Array [
        15,
        17,
      ],
      "name": "American Shorthair",
      "natural": true,
      "origin": "United States",
      "rare": false,
      "rex": false,
      "sheddingLevel": 3,
      "shortLegs": false,
      "socialNeeds": 4,
      "strangerFriendly": 3,
      "suppressedTail": false,
      "temperament": Array [
        "Active",
        "Curious",
        "Easy Going",
        "Playful",
        "Calm",
      ],
      "vocalisation": 3,
      "weight": Array [
        4,
        7,
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
